packages/manager/modules/pci/src/projects/project/storages/databases/database/general-information/general-information.controller.js

```diff
@@ -182,21 +190,34 @@ export default class {
   deleteDatabase() {
     this.trackDashboard('general_information::delete_database');
 
-    if (
-      [DATABASE_TYPES.KAFKA, DATABASE_TYPES.KAFKA_MIRROR_MAKER].includes(
-        this.database.engine,
-      )
-    ) {
+    if (isFeatureActivated('serviceIntegrationTab', this.database.engine)) {
       return this.DatabaseService.getIntegrations(
         this.projectId,
         this.database.engine,
         this.database.id,
       ).then((integrations) => {
-        const linkedServices = integrations.map((integration) =>
-          this.database.engine === DATABASE_TYPES.KAFKA
-            ? find(this.databases, { id: integration.destinationServiceId })
-            : find(this.databases, { id: integration.sourceServiceId }),
-        );
+        const linkedServices = integrations.reduce((acc, curr) => {
+          // if source equals destination, then the service is linked to himself
+          // this is the only scenario where we want to display it in the linked service
+          if (curr.sourceServiceId === curr.destinationServiceId) {
+            if (!acc.find((service) => service.id === curr.sourceServiceId)) {
+              acc.push(find(this.databases, { id: curr.sourceServiceId }));
+            }
+          }
+          if (
+            this.database.id !== curr.sourceServiceId &&
+            !acc.find((service) => service.id === curr.sourceServiceId)
+          ) {
+            acc.push(find(this.databases, { id: curr.sourceServiceId }));
+          }
+          if (
+            this.database.id !== curr.destinationServiceId &&
+            !acc.find((service) => service.id === curr.destinationServiceId)
+          ) {
+            acc.push(find(this.databases, { id: curr.destinationServiceId }));
+          }
+          return acc;
+        }, []);
```

Comment:
This `reduce` can be simplified by using a `Set` as it does not allow the same reference to be included twice
```javascript
const linkedServices = Array.from(integrations.reduce((databases, { sourceServiceId, destinationServiceId }) => {
  if (sourceServiceId === destinationServiceId || this.database.id !== sourceServiceId) {
    databases.add(this.databases.find(({ id }) => id === sourceServiceId));
  } else if (this.database.id !== curr.destinationServiceId) {
    databases.add(this.databases.find(({ id }) => id === destinationServiceId));
  }
  return databases;
}, new Set()));
```

Reply:
Done here: https://github.com/ovh/manager/commit/ab8e0199587826e2f4dd2ce1f47bbe4086b79303